### ISSUE TITLE
webui becomes unresponsive when trying to restore when a deleted client exists

### ISSUE THREAD

**Issue by sebsura** [bareos/bareos#1851]
### Bareos component version

24.0.0 (master)

### Steps to reproduce

1. Run a backup job on client 1
2. change name of client 1 to client 2 in the configuration
3. reload the configuration
4. in webui restore select client 2

### Expected results

I can select both client 1 and client 2 as source client, and selecting them does what one expects (i.e. you can select  jobs)

### Actual results

You can only select client 2 and doing so causes the webui to become unresponsive.
![Screenshots_59](https://github.com/bareos/bareos/assets/124262655/1672f961-f302-4e9a-a3f3-2d26fd183b8b)
[trace.zip](https://github.com/user-attachments/files/15738411/trace.zip)

### Relevant log output

```shell
Firefox console:

NS_ERROR_NOT_IMPLEMENTED: Component returned failure code: 0x80004001 (NS_ERROR_NOT_IMPLEMENTED) [nsIAppStartup.secondsSinceLastOSRestart]
    _collectStartupConditionsTelemetry resource:///modules/BrowserGlue.sys.mjs:1772
    BG__onFirstWindowLoaded resource:///modules/BrowserGlue.sys.mjs:1880
    BG_observe resource:///modules/BrowserGlue.sys.mjs:1102
    _delayedStartup chrome://browser/content/browser.js:2116
BrowserGlue.sys.mjs:1781:15
    _collectStartupConditionsTelemetry resource:///modules/BrowserGlue.sys.mjs:1781
    BG__onFirstWindowLoaded resource:///modules/BrowserGlue.sys.mjs:1880
    BG_observe resource:///modules/BrowserGlue.sys.mjs:1102
    _delayedStartup chrome://browser/content/browser.js:2116
TypeError: linkMap is undefined
    Links_onLinkChanged resource://gre/modules/NewTabUtils.sys.mjs:2003
    PlacesProvider__callObservers resource://gre/modules/NewTabUtils.sys.mjs:697
    handlePlacesEvents resource://gre/modules/NewTabUtils.sys.mjs:679
NewTabUtils.sys.mjs:699:19
    PlacesProvider__callObservers resource://gre/modules/NewTabUtils.sys.mjs:699
    handlePlacesEvents resource://gre/modules/NewTabUtils.sys.mjs:679
update.locale  file doesn't exist in either the application or GRE directories UpdateUtils.sys.mjs:143:13
    getLocale resource://gre/modules/UpdateUtils.sys.mjs:143
LoginManagerPrompter: Couldn't process supplied URIString: http://localhost:30036 6 LoginManagerPrompter.sys.mjs:999:16
Missing resource in locale de: devtools/client/toolbox.ftl
```
I created a trace and uploaded it as a zip in the results section (i cannot do so here)
The webui link ended in: 
```
restore/?jobid=&client=bareosfd2&restoreclient=&restorejob=&where=&pluginoptions=&fileset=&mergefilesets=1&mergejobs=1&limit=2000
```

**Comment by sebsura** [comment]
Looking further into it, it seems this is actually two different issues
1. the webui becomes unresponsive if the selected client does not have any jobs
2. i cannot restore from a client thats not part of the configuration anymore

**Comment by joergsteffens** [comment]
I started to work on this.

> Looking further into it, it seems this is actually two different issues
> 
>     1. the webui becomes unresponsive if the selected client does not have any jobs (BUG)

I fixed this.


>     2. i cannot restore from a client thats not part of the configuration (FEATURE REQUEST)

I started to implement this, however get surprised that `list backups client=<deleted_client>` and even `list jobs client=<deleted_client>` only returns `invalid client parameter` (however, the `restore` command can restore from deleted clients).
Implementing the required functionality for the webui would require to extend the bconsole commands first.
